### PR TITLE
drush process-meter is broken 

### PR DESCRIPTION
**Pull request by shahardo**
#252

### CODE REVIEW

File: client/app/scripts/services/charUsage.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ angular.module('negawattClientApp')
       var twoYearsBack = Math.floor(Date.now() / 1000) - 2 * 365 * 24 * 60 * 60;
 
       var filters = {
-        type: 'month',
+        type: '2'/*month*/,
```
Comment (amitaibu) [inline client/app/scripts/services/charUsage.js]: ``// Month`` above

File: negawatt/modules/custom/negawatt_normalizer/negawatt_normalizer.drush.inc [hunk 2]
```diff
@@ -69,13 +78,8 @@ function negawatt_server_query_electricity_last_processed_condition_alter(QueryA
  *   - time: Include only meters that their last-processed field is EARLIER
  *     than this value. Defaults to REQUEST_TIME.
  */
-function negawatt_normalizer_process_meters($meter_ids = array(), $options = array()) {
-  // Set default options.
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/negawatt_normalizer.drush.inc]: keep this.

File: negawatt/modules/custom/negawatt_normalizer/negawatt_normalizer.drush.inc [hunk 3]
```diff
@@ -118,17 +123,27 @@ function negawatt_normalizer_process_meters($meter_ids = array(), $options = arr
   }
 
   // Add sort to base query.
-  $base_query
-    ->fieldOrderBy('field_last_processed', 'value')
-    ->fieldOrderBy('field_meter_processed', 'value');
+//  $base_query
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/negawatt_normalizer.drush.inc]: remove.

File: negawatt/modules/custom/negawatt_normalizer/negawatt_normalizer.drush.inc [hunk 4]
```diff
@@ -149,8 +166,10 @@ function negawatt_normalizer_process_meters($meter_ids = array(), $options = arr
  *
  * @param array $meter_ids
  *   Array of meter-ids to process.
+ * @param array $time_frame
+ *   Array of from and to timestamps, giving the time frame to process.
  */
-function negawatt_normalizer_process_meters_once($meter_ids = array()) {
+function negawatt_normalizer_process_meters_once($meter_ids = array(), $time_frame = array()) {
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/negawatt_normalizer.drush.inc]: consider improving the name.

File: negawatt/modules/custom/negawatt_normalizer/negawatt_normalizer.module [hunk 5]
```diff
@@ -104,16 +104,16 @@ function negawatt_normalizer_get_electricity_normalizer_handler($plugin_name) {
...
  * @return array
  *    Array of processed entities.
  */
-function negawatt_normalizer_process_meter($node, $timestamp = NULL) {
+function negawatt_normalizer_process_meter($node, $time_frame = array()) {
   $wrapper = entity_metadata_wrapper('node', $node);
   $plugin_name = str_replace('_meter', '', $wrapper->getBundle());
   $handler = negawatt_normalizer_get_electricity_normalizer_handler($plugin_name);
-  return $handler->process($wrapper->value(), $timestamp);
+  return $handler->process($wrapper->value(), array()/*frequencies*/, $time_frame);
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/negawatt_normalizer.module]: move comment above.

Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/negawatt_normalizer.module]: ``return $handler->process($wrapper->value(), $frequencies = array(), $time_frame);``

File: negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php [hunk 6]
```diff
@@ -173,6 +173,7 @@ public function getTimestampBeginning() {
         // If timestampEnd was set to exact hour, go back one hour, otherwise go back to beginning of hour.
         $hour = (0 == $date_info['minutes'] && 0 == $date_info['seconds']) ? ($date_info['hours'] - 1) : $date_info['hours'];
         $date_time->setTime($hour, 0, 0);
+//msg("gtb: tse=$this->timestampEnd, di:{$date_info['year']}-{$date_info['mon']}-{$date_info['mday']} {$date_info['hours']}:{$date_info['minutes']}:{$date_info['seconds']}, h=$hour, tsb=".$date_time->getTimestamp());
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php]: remove

File: negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php [hunk 7]
```diff
@@ -317,12 +318,18 @@ public function process($node, $frequencies = array(), $time_period = array()) {
     $allowed_frequencies = $this->getAllowedFrequencies();
     $frequencies = empty($frequencies) ? $allowed_frequencies : $frequencies;
 
+    // If not time period is given, use '-', otherwise use '<from-date>-<to-date>'.
+    $time_period_str = empty($time_period) ? '-' : ( date('Y-m-d H:i', $time_period[0]) . ',' . date('Y-m-d H:i', $time_period[1]) );
+
+    // Output debug message.
+    msg(' frequencies:[' . implode(',', frequencyToStr($frequencies)) . '], ' . "time-period:[$time_period_str].");
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php]: ``negwatt_debug()``

File: negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php [hunk 8]
```diff
@@ -330,20 +337,13 @@ public function process($node, $frequencies = array(), $time_period = array()) {
...
-    $time_period_str = empty($time_period) ? '-' : ( date('Y-m-d H:i', $time_period[0]) . ',' . date('Y-m-d H:i', $time_period[1]) );
-
-    // Output a message when running under drush.
-    if (drupal_is_cli()) {
-      drush_log(' frequencies:[' . implode(',', $this->frequencyToStr($frequencies)) . '], '
-        . 'time-period:[' . $time_period_str . ']. ');
-      drush_log(' ' . count($processed_entities) . ' entities returned.');
-    }
+    // Output debug message.
+    msg(' ' . count($processed_entities) . ' entities returned.');
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php]: ``format_string('@count entities retruned)``

File: negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php [hunk 9]
```diff
@@ -330,20 +337,13 @@ public function process($node, $frequencies = array(), $time_period = array()) {
...
 
     // Generate message
     // Prepare arguments.
     $arguments = array(
-      '@frequencies' => implode(',', $this->frequencyToStr($frequencies)),
+      '@frequencies' => implode(',', frequencyToStr($frequencies)),
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php]: keep in object as static ``self::frequencyToStr()``

File: negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php [hunk 10]
```diff
@@ -496,6 +505,26 @@ protected function processRawEntities($fromTimestamp, $toTimestamp) {
...
+   * For minute and hour, subtract the number of seconds from $this->timestampEnd.
+   * This is because when switching from winter to summer clock, using
+   * $date_time->setTime, as in getTimestampBeginning, will repeat the same hour again!
+   *
+   * @return int
+   *  The end-timestamp for next time slice.
+   */
+  protected function getNextTimeSlice() {
+    switch ($this->getFrequency()) {
+      case \ElectricityNormalizerInterface::MINUTE: return $this->getTimestampEnd() - 60;
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php]: return line below

File: negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php [hunk 11]
```diff
@@ -511,11 +540,19 @@ protected function processRawEntities($fromTimestamp, $toTimestamp) {
...
+    msg('  in processNormalizedEntities. frequency: ' . frequencyToStr($this->getFrequency()) . ", time-frame: $time_period_str");
+
     // Start from the last time-slice, and loop backwards for all time-slices in time period.
     $processed_entities = array();
     $prev_period_entities = array();
     $this->setTimestampEnd($toTimestamp);
     do {
+      // Output debug message.
+      $time_period_str = date('Y-m-d H:i', $this->getTimestampBeginning()) . '-' . date('Y-m-d H:i', $this->getTimestampEnd());
+      msg("   time-frame: $time_period_str");
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php]: ``$this::debugMessage($message, $time_from = NULL, $time_to = NULL)``

File: negawatt/modules/custom/negawatt_normalizer/negawatt_normalizer.drush.inc (hunk 4 above)
Comment (shahardo) [inline negawatt/modules/custom/negawatt_normalizer/negawatt_normalizer.drush.inc]: The name is OK. I added a comment to clarify it.

**Pull request closed without merging**